mozilla/blurts-server
# Preserve the user's locale with form submissions

# References: 
Jira: MNTOR-1316
Figma: N/A




# Description

The user's locale would only get stored for requests for text/html documents. This meant that other requests, such as a POST form submission when adding an additional email address, would not have the user's locale available, e.g. when sending an email to confirm an additional email address.

Only tagging @toufali for review, since he [initially landed it](https://github.com/mozilla/blurts-server/commit/8c40a0e120700970eae277b3835e110c8474e084#diff-c72a907ac323cd2f334ed0e2bd07d15ab62581c4753660c8a0d1c681b30be4b6), and I'm not 100% confident about my changes here.

# How to test

Set your browser Accept-Language to a localised one (e.g. `fr-FR`). Create a new account, and check that the email you receive is in your chosen language. Then go to the Settings page and add another email address. The email you then receive on that other email address should also be in your chosen language, rather than in English.

# Checklist (Definition of Done)
- [ ] I've added a unit test to test for potential regressions of this bug. - Seems like a bit of a hassle, but could give it a shot.
